### ISSUE TITLE
ZXingScannerView - Camera view is stucked to old captured image on navigation pop

### ISSUE THREAD

**Issue by Arsalwali** [Redth/ZXing.Net.Mobile#461]
I have created page with ZXingScannerView and it successfully scan the code on first page and return the result to other page. Issue arise when I tap on back navigation to reopen the camera scan view. It shows old static captured image instead of camera scanning view. Furthermore this behavior is ambiguous, sometimes camera view appears and it again scan the barcode successfully.

I am testing this on Andriod phone API version 19.

Xaml Code:

```
     <Grid x:Name="grid">
      <Grid.RowDefinitions>
        <RowDefinition Height="96" />
        <RowDefinition Height="*" />
      </Grid.RowDefinitions>


      <Label VerticalOptions="FillAndExpand" HorizontalOptions="FillAndExpand"
             VerticalTextAlignment="Center" HorizontalTextAlignment="Center" LineBreakMode="WordWrap"
             Grid.Row="0" Text="Scan barcode" />
      <zxing:ZXingScannerView Grid.Row="1" x:Name="scanner" OnScanResult="OnScanCompleted"/>
    </Grid>
```

CodeBehind:
```
        public QRScanPage()
        {
                InitializeComponent();
        }
        private void OnScanCompleted(ZXing.Result result)
        {
            Device.BeginInvokeOnMainThread(() =>
            {
                scanner.IsAnalyzing = false;
                ViewModel.NavigateToNextPage(result.Text);
            });
        }

        protected override void OnAppearing()
        {
            base.OnAppearing();
            scanner.IsScanning = true;
        }

        protected override void OnDisappearing()
        {
            scanner.IsScanning = false;
            base.OnDisappearing();
        }
```

**Comment by zzm388** [comment]
Exact same behavior here @Arsalwali. Testing on API version 23.
Any progress with it?

**Comment by riteshbxr** [comment]
Same Issue with me... Has anyone Found any Solution??

**Comment by riteshbxr** [comment]
Found the solution.... 

> Use Scanning=true only once... In Scanner View Constructor or in OnAppearing of the Page..

```
 _zxing = new ZXingScannerView
            {
                VerticalOptions = LayoutOptions.Center,
                HorizontalOptions = LayoutOptions.Center,
                HeightRequest = 250,
                WidthRequest = 250,
                IsAnalyzing = true,
                IsScanning = true,

            };
```

> Don't write anything in OnDisappearing...
     
```
 protected override void OnDisappearing()
        {
           // _zxing.IsScanning = false;
            base.OnDisappearing();

        }
```


> IsAnalysing to be set false once scanning complete and should be set true in OnAppearing...

  ```
_zxing.OnScanResult += (result) =>
                 Device.BeginInvokeOnMainThread(async () =>
                 {
                     if (!string.IsNullOrWhiteSpace(result.Text))
                     {
                         _zxing.IsAnalyzing = false;
                         await OnGettingResult(result.Text);
                     }
                 });


protected override void OnAppearing()
        {

            base.OnAppearing();
            _zxing.IsAnalyzing = true;

           //Not required if already set while intialization
           //_zxing.IsScanning = true;

        }
```

**Comment by Redth** [comment]
Thanks for reporting this issue!  Unforunately it took me way too long to respond 😭.  Sorry, I apologize!  Recently the source code for this project was completely refactored to modernize it.  Many PR's were included in this effort, and many bugs were hopefully fixed.  Please try out the latest 3.x series of NuGet packages (currently in prerelease).  To try and make the project more maintainable in my spare time going forward, I've decided to close all existing issues to start with a clean slate.  If you're still experiencing this issue on the newest version, please open a new issue with as much detail as possible.  Thank you for your patience and understanding!  Happy scanning!
